examples/squiral/main.go

```diff
@@ -0,0 +1,341 @@
+// Copyright 2019 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build example jsgo
+
+// This demo is inspired by the xscreensaver 'squirals'.
+
+package main
+
+import (
+	"fmt"
+	"image/color"
+	"math/rand"
+	"time"
+
+	"github.com/hajimehoshi/ebiten"
+	"github.com/hajimehoshi/ebiten/ebitenutil"
+	"github.com/hajimehoshi/ebiten/inpututil"
+)
+
+const (
+	width         = 800
+	height        = 600
+	scale         = 1
+	numOfSquirals = width / 32
+)
+
+type palette struct {
+	name   string
+	colors []color.Color
+}
+
+var (
+	background = color.Black
+
+	palettes = []palette{
+		{
+			name: "sand dunes",
+			colors: []color.Color{
+				color.RGBA{0xF2, 0x74, 0x05, 0xFF}, //#F27405
+				color.RGBA{0xD9, 0x52, 0x04, 0xFF}, //#D95204
+				color.RGBA{0x40, 0x18, 0x01, 0xFF}, //#401801
+				color.RGBA{0xA6, 0x2F, 0x03, 0xFF}, //#A62F03
+				color.RGBA{0x73, 0x2A, 0x19, 0xFF}, //#732A19
+			},
+		},
+		{
+			name: "mono desert sand",
+			colors: []color.Color{
+				color.RGBA{0x7F, 0x6C, 0x52, 0xFF}, //#7F6C52
+				color.RGBA{0xFF, 0xBA, 0x58, 0xFF}, //#FFBA58
+				color.RGBA{0xFF, 0xD9, 0xA5, 0xFF}, //#FFD9A5
+				color.RGBA{0x7F, 0x50, 0x0F, 0xFF}, //#7F500F
+				color.RGBA{0xCC, 0xAE, 0x84, 0xFF}, //#CCAE84
+			},
+		},
+		{
+			name: "land sea gradient",
+			colors: []color.Color{
+				color.RGBA{0x00, 0xA2, 0xE8, 0xFF}, //#00A2E8
+				color.RGBA{0x67, 0xA3, 0xF5, 0xFF}, //#67A3F5
+				color.RGBA{0xFF, 0xFF, 0xD5, 0xFF}, //#FFFFD5
+				color.RGBA{0xDD, 0xE8, 0x0C, 0xFF}, //#DDE80C
+				color.RGBA{0x74, 0x9A, 0x0D, 0xFF}, //#749A0D
+			},
+		},
+	}
+	selectedPalette = 0
+	colorCycle      = 0
+	canvas          *ebiten.Image
+	auto            *automaton
+	blocker         = color.RGBA{0, 0, 0, 254}
+	free            = color.RGBA{0, 0, 0, 0}
+
+	rotors = [2][3]int{
+		[3]int{1, 0, 3},
+		[3]int{3, 0, 1},
+	}
+
+	// east, south, west, north
```

Comment:
Please follow a standard Go guideline. The comment should start with `dirs is` in this case.